Elements/src/Ceiling.cs

```diff
@@ -0,0 +1,188 @@
+        public Polygon Perimeter { get; protected set; }
+
+        /// <summary>
+        /// A collection of openings in the ceiling.
+        /// </summary>
+        public List<Opening> Openings { get; } = new List<Opening>();
+
+        /// <summary>
+        /// Construct a ceiling by extruding a profile.
+        /// </summary>
+        /// <param name="perimeter">The plan profile of the ceiling. It should lie on XY plane.
+        /// The Z coordinate will be ignored</param>
+        /// <param name="thickness">The thickness of the ceiling.</param>
+        /// <param name="elevation">The elevation of the ceiling.</param>
+        /// <param name="material">The material of the ceiling.</param>
+        /// <param name="transform">An optionional transform for the ceiling.</param>
+        /// <param name="representation">The ceiling's representation.</param>
+        /// <param name="isElementDefinition">Is this an element definition?</param>
+        /// <param name="id">The id of the ceiling.</param>
+        /// <param name="name">The name of the ceiling.</param>
+        [JsonConstructor]
+        public Ceiling(Polygon perimeter,
+                      double thickness,
+                      double elevation,
+                      Material material = null,
+                      Transform transform = null,
+                      Representation representation = null,
+                      bool isElementDefinition = false,
+                      Guid id = default(Guid),
+                      string name = null) : base(transform != null ? transform : new Transform(),
+                                                 material != null ? material : BuiltInMaterials.Concrete,
+                                                 representation != null ? representation : new Representation(new List<SolidOperation>()),
+                                                 isElementDefinition,
+                                                 id != default(Guid) ? id : Guid.NewGuid(),
+                                                 name)
+        {
+            if (thickness <= 0.0)
+            {
+                throw new ArgumentOutOfRangeException("The ceiling could not be created. The thickness of the ceiling must be greater than 0.0.");
+            }
+
+            if (!perimeter.Normal().IsParallelTo(Vector3.ZAxis))
+            {
+                throw new ArgumentOutOfRangeException("The ceiling could not be created. The perimeter polygon must lie on XY plane");
+            }
+
+            this.Elevation = elevation;
+            this.Perimeter = perimeter.Project(new Plane(Vector3.Origin, Vector3.ZAxis)); ;
+            this.Thickness = thickness;
+            Transform.Move(Vector3.ZAxis * Elevation);
+        }
+
+        /// <summary>
+        /// Construct a ceiling by extruding a profile.
+        /// </summary>
+        /// <param name="perimeter">The plan perimeter of the ceiling. It should lie on XY plane.
```

Comment:
"...should lie on _the_ XY plane."

Because this is required (we throw an Exception if it's not), we could say "must lie in the XY plane"